state-chain/runtime/src/chainflip/decompose_recompose.rs

```diff
@@ -20,25 +15,20 @@ fn select_median<T: Ord + Copy>(mut data: Vec<T>) -> Option<T> {
 fn decode_many<T: Encode + Decode>(data: &mut [Vec<u8>]) -> Vec<T> {
 	data.iter_mut()
 		.map(|entry| T::decode(&mut entry.as_slice()))
+		.inspect(|r|
+			if let Err(e) = r {
+				log::error!(
+					"Error decoding {}: {}.\nThis is expected to happen right after the runtime-upgrade, but should only be observed until the end of the epoch (PRO-1073)",
```

Comment:
What was the issue with migrating the ExtraCallData?

We know that ExtraCallData is always an old BitcoinFeeInfo in the Bitcoin instance of the pallet, and we know that we can decode it using decode_many and passing the old type in, then we can do the same `undo` as written in the other migration for each of these values, and then write it back ?

Reply:
`ExtraCallData` is a storage of the `witnesser` pallet.
The `witnesser` pallet does not seem to be separated in a way as `chain-traking` is.

I might be wrong though.

Reply:
Ah yes, ofc, you're right. hmm